Vic2ToHoI4/Source/HOI4World/HoI4Country.cpp

```diff
@@ -874,4 +874,71 @@ std::optional<HoI4FocusTree> HoI4::Country::getNationalFocus() const
 	{
 		return nullopt;
 	}
+}
+
+void HoI4::Country::addSphereling(std::string Sphereling)
+{
+	double defaultAutonomy = 0.5;
+	spherelings.insert(make_pair(Sphereling, defaultAutonomy));
+}
+
+double HoI4::Country::getSpherelingAutonomy(std::string spherelingTag) const
+{
+	auto sphereling = spherelings.find(spherelingTag);
+	if (sphereling == spherelings.end())
+	{
+		LOG(LogLevel::Debug) << "Could not find " << sphereling->first << " in spherelings";
+		return 0.5;
+	}
+	else
+	{
+		return sphereling->second;
+	}
+}
+
+void HoI4::Country::setSpherelingAutonomy(std::string spherelingTag, double autonomy)
+{
+	auto sphereling = spherelings.find(spherelingTag);
+	if (sphereling == spherelings.end())
+	{
+		LOG(LogLevel::Debug) << "Could not find " << sphereling->first << " in spherelings";
+		sphereling->second = 0.5;
+	}
+	else
+	{
+		sphereling->second = autonomy;
+	}
+}
+
+double HoI4::Country::calculateInfluenceFactor()
+{
+	auto GPInfluences = getGPInfluences();
```

Comment:
No need for this, you can just use the class member directly.

Reply:
Acknowledged